errai-codegen/src/main/java/org/jboss/errai/codegen/control/SwitchBlock.java

```diff
@@ -86,16 +87,20 @@ public String generate(Context context) {
     checkSwitchExprType();
 
     if (!caseBlocks.isEmpty()) {
+      final MetaClass switchExprMetaClass = switchExprStmt.getType().getErased().asBoxed();
+      final boolean isStringSwitch = switchExprMetaClass.asClass().equals(String.class);
       for (final LiteralValue<?> value : caseBlocks.keySet()) {
-        if (!switchExprStmt.getType().getErased().asBoxed().isAssignableFrom(value.getType().getErased())) {
+        if (!switchExprMetaClass.isAssignableFrom(value.getType().getErased())) {
           throw new InvalidTypeException(
               value.generate(context) + " is not a valid value for " + switchExprStmt.getType().getFullyQualifiedName());
         }
-        // case labels must be unqualified
         String val = value.generate(context);
-        final int idx = val.lastIndexOf('.');
-        if (idx != -1) {
-          val = val.substring(idx + 1);
+        if (!isStringSwitch) {
```

Comment:
So if I am understanding this part correctly, this fixes possibly buggy behaviour when the case label is a string containing dots?